File: Core/TFX.Core.UI/src/Model/XServiceModel.ts

interface XColumnModel
{
    Name: string;
    Visible: boolean;
    Width: number;
    Title: string;
    Align: XAlign;
    Mask: string;
    IsFreeSearch: boolean;
    Operator: XOperator;
    MaxLenght: number;
    Type: string;
    FieldID: string;
}

interface XDataViewModel
{
    Columns: XColumnModel[]
}

interface XServiceModel
{
    ID: string;
    Forms: XFRMModel[];
    DataView: XDataViewModel;
    PKFieldName: string
    SearchPath: string;
    GetPath: string;
    FlushPath: string;
}

File: Core/TFX.Core/Model/Service/XServiceModel.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Microsoft.EntityFrameworkCore.Update.Internal;

using TFX.Core.DB;
using TFX.Core.Model.FRM;
using TFX.Core.Services;

namespace TFX.Core.Model.Service
{

    public class XColumnModel
    {
        public string Name
        {
            get; set;
        }
        public string Description
        {
            get; set;
        }
        public string Type
        {
            get; set;
        }
        public string Title
        {
            get; set;
        }
        public string Value
        {
            get; set;
        }
        public string Mask
        {
            get; set;
        }
        public int MaxLenght
        {
            get; set;
        }
        public bool Visible
        {
            get; set;
        }
        public bool IsFreeSearch
        {
            get; set;
        }
        public XOperator Operator
        {
            get; set;
        }

    }

    public class XDataViewModel
    {
        public XDataViewModel()
        {
            Columns = [];
        }

        public List<XColumnModel> Columns
        {
            get; set;
        }
    }


    public class XServiceModel
    {
        public XServiceModel()
        {
            DataView = new XDataViewModel();
            Forms = [];
        }

        public string SearchPath
        {
            get;
            set;
        }
        public List<XFRMModel> Forms
        {
            get;
        }

        public XDataViewModel DataView
        {
            get;
        }
    }
}
